example/spec/ingestGranule/IngestUMMGSuccessSpec.js

```diff
@@ -427,21 +419,28 @@ describe('The S3 Ingest Granules workflow configured to ingest UMM-G', () => {
     it('updates the UMM-G JSON file in S3 with new paths', async () => {
       const updatedUmm = await getUmmObject(newS3UMMJsonFileLocation);
 
-      const relatedUrlDifferences = updatedUmm.RelatedUrls.filter((urlObject) => {
-        // Skip non-science URLs and public S3 URLs
-        if (!isUMMGScienceUrl(urlObject.URL) ||
-            urlObject.URL.match(/s3\.amazonaws\.com/)) {
-          return false;
-        }
-        const relatedUrl = new URL(urlObject.URL);
-        relatedUrl.host = process.env.DISTRIBUTION_ENDPOINT;
-        return !originalUmmUrls.includes(relatedUrl.toString());
-      });
+      const expectedToChange = updatedUmm.RelatedUrls
+        .filter((urlObject) => urlObject.URL.match(/.*.hdf$/))
+        .map((urlObject) => urlObject.URL);
+      const expectedSame = updatedUmm.RelatedUrls
+        .filter((urlObject) => !urlObject.URL.match(/.*.hdf$/))
+        .map((urlObject) => urlObject.URL);
 
       // Only the file that was moved was updated
-      expect(relatedUrlDifferences.length).toEqual(1);
-
-      expect(relatedUrlDifferences[0].URL).toContain(destinationKey);
+      expect(expectedToChange.length).toEqual(1);
+      expect(expectedToChange[0]).toContain(destinationKey);
+
+      const unchangedOriginals = originalUmmUrls.filter((original) => !original.match(/.*.hdf$/));
+      expect(unchangedOriginals.length).toEqual(expectedSame.length);
+
+      // Each originalUmmUrl (removing the DISTRIBUTION_ENDPOINT) should be found
```

Comment:
I'm still having trouble following why this is the case. Is it because the API lambda automatically will have the "real" distribution API endpoint injected as an environment variable? What creates the "original" UMM JSON and where it does get the `DISTRIBUTION_ENDPOINT` environment variable from?

Reply:
During the previous describes, we're setting process.env.DISTRIBUTION_ENDPOINT in setDistributionApiEnvVars(), and it's defaulting to `http://localhost:5002` In my stack deployment, I have the real api endpoint defined on the `mhs3-apiEndpoints` lambda which is `https://rz0l1ei1p1.execute-api.us-east-1.amazonaws.com/dev/` . The that was there before, wasn't actually testing it correctly, because it was just replacing host with localhost:5002, and that was generating a `http://localhost:5002/dev/bucket/key` url.  

I think the problem is that we don't have a way to know the real distribution_endpoint when we run the tests.  But I was wondering if we should expose it somewhere on the api.  Then we could get the correct endpoint at the start.

Reply:
ON this line: https://github.com/nasa/cumulus/pull/932/files#diff-76cc6b5bf7a5e3192ad0c3be727b0041L437

If you have ideas, I'm ears, because I wanted to talk through this earlier.

Reply:
Thanks for explaining. And yeah, we were doing the same thing before except using the `URL` library to replace the host. I don't think there's a way to get around this.

I'm not sure I understand how the previous test wasn't testing it correctly though